datafusion/expr/src/expr_fn.rs

```diff
@@ -843,18 +802,9 @@ impl WindowUDFImpl for SimpleWindowUDF {
     }
 }
 
-/// Calls a named built in function
-pub fn call_fn(name: impl AsRef<str>, args: Vec<Expr>) -> Result<Expr> {
```

Comment:
This is an interesting API change -- I wonder if this API is used to call functions by name in the expr API 🤔 

However, it is currently going to be missing all the built in functions now that we port them to UDFs so a compiler error is probably better than a runtime error, 

Maybe we can add such a function back eventually (using the API @jayzhan211  and @viirya  were talking about in https://github.com/apache/arrow-datafusion/issues/9892 🤔 )

Reply:
I think it's a useful function but from what I recall the compiler was complaining about it being unused currently. Of course, that doesn't mean it's not useful externally (it obviously is) but I think that re-implementing it in #9892 would be the best approach to take. I could add it back in during #10090 but I think it may have to live somewhere else unless a session context or something similar is passed in though

Reply:
> it may have to live somewhere else unless a session context or something similar is passed in though

Yes I agree with this. Now that there is no static global list of functions some sort of state needs to get passed in